src/main/kotlin/com/bantolomeus/service/BookService.kt

```diff
@@ -16,84 +16,87 @@ class BookService(private val bookRepository: BookRepository,
 
     fun createBook(bookDTO: BookDTO): BookDTO {
         val savedBookDTO = bookRepository.saveBookIfItNotExists(bookDTO)
-        if (savedBookDTO.currentPage > 0L) {
+        if (savedBookDTO.currentPage > 0) {
             saveBookUpdate(savedBookDTO)
         }
         return bookDTO
     }
 
     fun updateBook(bookUpdate: BookUpdateInputDTO, bookName: String): BookUpdatesFileDTO {
-        var response = BookUpdatesFileDTO()
-        var books = bookRepository.getBooks().books.filter { it.name != bookName }
-        var oldPage = Long.MAX_VALUE
-        val foundBook = bookRepository.getBookByName(bookName)
+        val foundBook = bookRepository.getBookByName(bookName) ?: return BookUpdatesFileDTO()
 
-        if (foundBook.name == bookName) {
-            foundBook.let {
-                oldPage = it.currentPage
-                it.currentPage = bookUpdate.currentPage
-                if (it.currentPage >= it.pagesTotal) {
-                    it.readTime = (Date().time - dateFormat.parse(it.dateStarted).time) / DIVISOR_FOR_DAY
-                }
-            }
-            books = mutableListOf(foundBook) + books.toMutableList()
+        if (foundBook.currentPage == foundBook.pagesTotal) {
+            foundBook.readTime = ((Date().time - dateFormat.parse(foundBook.dateStarted).time) / DIVISOR_FOR_DAY) + 1
+        }
+        val oldPage = foundBook.currentPage
+        foundBook.currentPage = bookUpdate.currentPage
+        val pagesRead = bookUpdate.currentPage.minus(oldPage)
+        if (pagesRead <= 0 || oldPage >= bookUpdate.currentPage || bookUpdate.currentPage > foundBook.pagesTotal) {
+            return BookUpdatesFileDTO()
         }
 
```

Comment:
Why is here space but not after the last if?

Reply:
with *last* you actually mean *previous* ? So the first if in this `updateBook` method belongs to the precondition checks that are happening in the method. I tried to move all precondition checks into one block, separate this block with a newline and then put the actual business/update logic. So you can skip easily from the preconditions to the business logic and don't have too many confusing blank linkes..

Does that make sense?

Reply:
Dunno. Lets keep it and change it if it annoys someone in the future.